### PR TITLE
Fetch already applied mercurial patches for git

### LINKED ISSUE #1506: Instead of applying patch in git, fetch it from the local Mercurial repository
Using git-cinnabar, we can replace https://github.com/mozilla/bugbug/blob/e60a51b6b744ae1c1bbd65cbc28fc7f796f851e0/scripts/commit_classifier.py#L358-L380 and just fetch the commit from the local Mercurial repository where we already applied the patch.

Then, we can stop installing `patch` in our Docker image: https://github.com/mozilla/bugbug/blob/e60a51b6b744ae1c1bbd65cbc28fc7f796f851e0/infra/dockerfile.commit_retrieval#L10.

### PR DESCRIPTION
**Pull request by KhadijaKamran**
Closes #1506
Sorry if the change is completely wrong. I wanted to start a PR for further discussion.

### CODE REVIEW

File: scripts/commit_classifier.py [hunk 1]
```diff
@@ -381,7 +381,7 @@ def load_user(phid):
 
             if self.git_repo_dir:
                 patch_proc = subprocess.Popen(
-                    ["patch", "-p1", "--no-backup-if-mismatch", "--force"],
+                    ["git", "cinnabar", "hg2git", patch.phid],
```
Comment (suhaibmujahid) [inline scripts/commit_classifier.py:384]: `patch.phid` is not a changeset, it is a Phabricator ID. 

The command is searching for a git commit that is corresponding to the provided hg changeset. This is not what we want to do. The goal here is to fetch the changeset from the local Mercurial repository. See https://github.com/glandium/git-cinnabar#fetching-a-specific-mercurial-changeset.

**Review by suhaibmujahid — COMMENTED** [review]
Thank you for working on this, @KhadijaKamran!

File: scripts/commit_classifier.py [hunk 2]
```diff
@@ -381,7 +381,13 @@ def load_user(phid):
 
             if self.git_repo_dir:
                 patch_proc = subprocess.Popen(
-                    ["patch", "-p1", "--no-backup-if-mismatch", "--force"],
+                    [
+                        "git",
+                        "cinnabar",
+                        "fetch",
+                        f"hg::{self.repo_dir}",
+                        diffs[patch.phid],
```
Comment (suhaibmujahid) [inline scripts/commit_classifier.py:389]: Did you run the code? Did it do what you expect? 

`diffs` is a list and `patch.phid` is a string, so this should raise a type error. What we need is hash of the changeset that we just applied to the local Mercurial repository.


BTW. The cod that is applying the patch in git need to be deleted.

Please run the code and ensure it is doing what you expect.

Comment (KhadijaKamran) [reply scripts/commit_classifier.py:389]: Sorry, I was of the understanding that diffs is a dictionary. Is changeset actually diff?

Comment (KhadijaKamran) [reply scripts/commit_classifier.py:389]: Hey @suhaibmujahid 
Sorry, I did not run the code. I was confused as to what arguments I should pass in order to run it.

Comment (KhadijaKamran) [reply scripts/commit_classifier.py:389]: I have read the code in detail and tried to understand it best. I understand how patches are applied to the local mercurial repository. I am still confused as to where can I find the changeset. Is it available in the `patch`? Kindly help me with this @suhaibmujahid.

Comment (suhaibmujahid) [reply scripts/commit_classifier.py:389]: > I am still confused as to where can I find the changeset. 

The changeset is what we applied here:

https://github.com/mozilla/bugbug/blob/f9906057a5281b8913fb5a92edbe73440953581b/scripts/commit_classifier.py#L376-L380

What you need in your command is the changeset ID/hash that we just applied.

**Comment by suhaibmujahid** [comment]
@KhadijaKamran are you still interested in working on this?

**Comment by suhaibmujahid** [comment]
I'm closing the PR for lack of response; please feel free to reopen if you still want to work on this.

**Pull request closed without merging**